lua/telescope/finders/async_function_finder.lua

```diff
@@ -0,0 +1,53 @@
+local async_lib = require "plenary.async_lib"
+local async = async_lib.async
+local await = async_lib.await
+local void = async_lib.void
+
+local make_entry = require "telescope.make_entry"
+
+return function(opts)
+  local results_maker
+  local entry_maker
+  if type(opts) == 'function' then
+    results_maker = opts
+    entry_maker = make_entry.gen_from_string()
+  else
+    results_maker = opts.results_maker
+    entry_maker = opts.entry_maker or make_entry.gen_from_string()
+  end
+
+
+  local function make_results()
+    local input_results = results_maker()
+    local results = {}
+    for k, v in ipairs(input_results) do
+      local entry = entry_maker(v)
+
+      if entry then
+        entry.index = k
+        table.insert(results, entry)
+      end
+    end
+    return results
+  end
+
+  return setmetatable({
+    results = {'results'},  -- TODO
```

Comment:
Not 100% sure, but could you please check what `results` in `:lua print(vim.inspect(TelescopeGlobalState[vim.api.nvim_get_current_buf()].picker.finder))` looks like? Likely just `{'results'}`. 

If so, would 
```lua
    results = {}
    close = function() end,
  }, {
    __call = void(async(function(_, _, process_result, process_complete)
      results = make_results()
```

Fix it? The idea is `finder.results` should hold the table, though probably _in practice_ doesn't matter as results are properly hand over to the `EntryManager`.

Reply:
I'll try it but would that actually work since is that actually the same `results`? I haven't worked much with metatables before but is there a notion of "self"? Ie if it should be something like `self.results = make_results()`?

Reply:
Ah just read now that the first argument is the table itself, fixed in latest commit

Reply:
Yep, you are right. Elegant solve :)